internal/sourceanalysis/go.go

```diff
@@ -1,35 +1,77 @@
 package sourceanalysis
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
 	"fmt"
-	"path/filepath"
+	"io"
+	"log"
+	"os"
 
-	"github.com/google/osv-scanner/internal/govulncheckshim"
+	"github.com/google/osv-scanner/internal/sourceanalysis/govulncheck"
 	"github.com/google/osv-scanner/pkg/models"
-	"github.com/google/osv-scanner/pkg/reporter"
 	"golang.org/x/exp/slices"
-	"golang.org/x/vuln/exp/govulncheck"
+	"golang.org/x/vuln/scan"
 )
 
-func goAnalysis(r reporter.Reporter, pkgs []models.PackageVulns, source models.SourceInfo) {
-	vulns, vulnsByID := vulnsFromAllPkgs(pkgs)
-	res, err := govulncheckshim.RunGoVulnCheck(filepath.Dir(source.Path), vulns)
+func goAnalysis(dir string, pkgs []models.PackageVulns) (_ []models.PackageVulns, err error) {
+	_, vulnsByID := vulnsFromAllPkgs(pkgs)
+
+	dbdir, err := os.MkdirTemp("", "")
 	if err != nil {
-		// TODO: Better method to identify the type of error and give advice specific to the error
-		r.PrintError(
-			fmt.Sprintf("Failed to run code analysis (govulncheck) on '%s' because %s\n"+
-				"(the Go toolchain is required)\n", source.Path, err.Error()))
+		log.Fatal(err)
+	}
+	defer func() {
+		rerr := os.RemoveAll(dir)
+		if err == nil {
+			err = rerr
+		}
+	}()
 
-		return
+	cmd := scan.Command(context.Background(), "-db", fmt.Sprintf("file://%s", dbdir), "-C", dir, "-json", "./...")
```

Comment:
Can you add a comment for aht this is doing?

Reply:
Done in https://github.com/google/osv-scanner/pull/431/files